### PR TITLE
Update e2e workflow to be broader in scope

### PR DESCRIPTION
**Pull request by pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

**Description**

Updates the Meshery e2e test workflow which uses `mesheryctl`.

Current status of the workflow:

* Runs tests on multiple Kubernetes versions
* Runs tests in and out of cluster
* Runs performance tests
* Deploys service mesh
* Applies service mesh patterns (3 now)
* Scheduled to run twice daily

More tests can be added subsequently. An example of this workflow run can be found [here](https://github.com/navendu-pottekkat/action-tests/actions/runs/1603683921).

- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — APPROVED** [review]

File: .github/sample-test-config.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      load_generator: fortio
+      protocol: 1
+      connections: 2
+      rps: 10
+      headers: {}
+      cookies: {}
+      body: ''
+      content_type: ''
+      endpoint_urls:
+        - 'https://www.cncf.io/'
```
Comment (leecalcote) [inline .github/sample-test-config.yaml:22]: Let's change this to meshery.io

Comment (navendu-pottekkat) [reply .github/sample-test-config.yaml:22]: Didn't want to cause any issues with this traffic 😅

Comment (navendu-pottekkat) [reply .github/sample-test-config.yaml:22]: Will make the change if needed in a subsequent PR with more tests.

Comment (leecalcote) [reply .github/sample-test-config.yaml:22]:
```suggestion
        - 'https://meshery.io'
```

File: .github/sample-test-config.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Test configuration file for running performance benchmarks
```
Comment (leecalcote) [inline .github/sample-test-config.yaml:1]: Please change the name of this file to something other than "sample". If this is being used for integration tests, please indicate this.

File: .github/workflows/mesheryctl-e2e.yaml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Run Performance Tests
+        uses: layer5io/meshery-smp-action@master
+        with:
+          provider_token: ${{ secrets.PROVIDER_TOKEN }}
+          platform: ${{ matrix.platform }}
+          profile_name: sample-e2e-workflow
```
Comment (leecalcote) [inline .github/workflows/mesheryctl-e2e.yaml:111]: sample? are the e2e flows being used? Call them that, then.

File: .github/workflows/mesheryctl-e2e.yaml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        required: false
+      patternfile_url:
+        description: "URL of the pattern file"
+        required: false
+      patternfile_name:
+        description: "name of the pattern file"
+        required: false
+  # scheduled to run at the 43rd minute of every 12th hour
+  schedule:
+    - cron: '43 */12 * * *'
```
Comment (leecalcote) [inline .github/workflows/mesheryctl-e2e.yaml:19]:
```suggestion
    - cron: '43 */24 * * *'
```

Out of the 2,000 available a month, do you know how many build minutes this eats up? Let's start with once-a-day.

File: .github/sample-test-config.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+# See: https://docs.meshery.io/guides/performance-management#running-performance-benchmarks-through-mesheryctl
+test:
+  smp_version: v0.0.1
+  
+  # The name of the test
+  name: Sample Performance Test
```
Comment (leecalcote) [inline .github/sample-test-config.yaml:7]:
```suggestion
  name: Performance Profile for Nightly Meshery Integration Testing
```

File: .github/sample-test-config.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      protocol: 1
+      connections: 2
+      rps: 10
+      headers: {}
+      cookies: {}
+      body: ''
+      content_type: ''
+      endpoint_urls:
+        - 'https://meshery.io'
+  duration: 2s
```
Comment (leecalcote) [inline .github/sample-test-config.yaml:23]:
```suggestion
  duration: 10s
```

File: .github/workflows/mesheryctl-e2e.yaml (hunk 3 above)
Comment (leecalcote) [reply .github/workflows/mesheryctl-e2e.yaml:111]:
```suggestion
          profile_name: integration-e2e-workflow
```

File: .github/workflows/mesheryctl-e2e.yaml [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Run Performance Tests
+        uses: layer5io/meshery-smp-action@master
+        with:
+          provider_token: ${{ secrets.PROVIDER_TOKEN }}
+          platform: ${{ matrix.platform }}
+          profile_name: sample-e2e-workflow
+          profile_filename: sample-test-config.yaml
```
Comment (leecalcote) [inline .github/workflows/mesheryctl-e2e.yaml:112]:
```suggestion
          profile_filename: integration-test-config.yaml
```

File: .github/integration-test-config.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
+# Test configuration file for running performance benchmarks
+# See: https://docs.meshery.io/guides/performance-management#running-performance-benchmarks-through-mesheryctl
+test:
+  smp_version: v0.0.1
+  
+  # The name of the test
+  name: Sample Performance Test
```
Comment (leecalcote) [inline .github/integration-test-config.yaml:7]:
```suggestion
  name: Performance Profile for Meshery's Nightly Integration Testing
```

**Comment by leecalcote** [comment]
@Navendu the forthcoming, nightly Meshery integration tests can publish results into this report as well --> https://docs.meshery.io/project/compatibility-matrix

**Pull request merged**